deps/build.jl

```diff
@@ -1,65 +1,58 @@
-function find_prev_tag(tag)
-    project_root = (@__DIR__) |> dirname |> abspath
-    run(`git -C $project_root fetch --tags`)
-    tags = readlines(`git -C $project_root tag`)
-    sort!(tags)
-    idx = indexin([tag], tags)[1]
-    if idx == nothing return "NO-PREV-TAG" end
-    return get(tags, idx - 1, "NO-PREV-TAG")
-end
+###
+### This file is generated by running
+### ` julia generate_buildjl.jl L/Libtask/build_tarballs.jl`
+### in the Yggdrasil root directory, with 2 updates:
+### 1. add prefix tp products, see https://github.com/JuliaPackaging/Yggdrasil#binaryproviderjl,
+### 2. products filter
+###
+
+using BinaryProvider # requires BinaryProvider 0.3.0 or later
+
+# Parse some basic command-line arguments
+const verbose = "--verbose" in ARGS
+const prefix = Prefix(get([a for a in ARGS if a != "--verbose"], 1, joinpath(@__DIR__, "usr")))
+products = [
+    LibraryProduct(prefix, ["libtask_v1_0"], :libtask_v1_0),
+    LibraryProduct(prefix, ["libtask_v1_1"], :libtask_v1_1),
+    LibraryProduct(prefix, ["libtask_v1_2"], :libtask_v1_2),
+    LibraryProduct(prefix, ["libtask_v1_3"], :libtask_v1_3),
+]
 
-# modify build-tmp.jl to only check correct version libs
-function install_products_filter(build_file)
-    prod_filter = raw"""products_tmp = filter(products) do prod
+products_tmp = filter(products) do prod
     endswith(prod.libnames[1], "$(VERSION.major)_$(VERSION.minor)")
 end
 length(products_tmp) == 0 && (products_tmp = [products[end]])
 products = products_tmp
```

Comment:
Looks good, I guess we don't have to care about version jumps? In that case, something like
```suggestion
lastminor = findlast(products) do p
    x = match(r"v(\d)_(\d)$", string(p.variable_name))
    x !== nothing && VERSION.major >= parse(Int, x.captures[1]) && VERSION.minor >= parse(Int, x.captures[2])
end
products = products[[lastminor]]
```
would provide a the last compatible version.